docs/contributing/guides/release-checklist.md

```diff
@@ -0,0 +1,85 @@
+---
+sidebar_position: 3
+---
+
+# Release Checklist
+
+## :bust_in_silhouette: Developer
+
+### :hatching_chick: Create an End-of-Release Review
+
+1. Create a new branch from `develop`.
+
+![Creating a Branch for Review](/images/release-branch.png)
+![Naming the review branch](/images/release-branch-name.png)
+
+2. Create a Pull Request from your review branch to `main`.
+
+![Creating a PR for Review](/images/release-pr.png)
+
+This method allows the developer and reviewer to see all changes since the previous release.
+
+Why not create a `develop` -> `main` PR for the review?
+- Peer reviewers may request code changes.
+- `develop` is a protected branch that can only be pushed into through a PR.
+- Using an unprotected branch `r#-final-review` allows direct pushes to address review comments.
+
+### :mag: Hold Review
+
+1. Inform peer reviewer(s) and await review comments.
+2. Push review fixes to your `r#-final-review` branch.
+3. After approval(s), when your code is passing all CI checks, you may merge to `main`.
+
+### :broom: Cleanup
+
+1. Since `develop` also needs the changes added to `r#-final-review`, open a PR to merge `main` into `develop`.
+2. Have the same reviewers approve the PR.
+
+## :busts_in_silhouette: Peer Reviewers
+
+This is the *last chance* for a review of the code before it is immortalized forever as a new release.
+
+Some checks aren't currently automated.
+
+Please paste the following comment template into the review:
+```
+## Documents
+- [ ] ICD
+  - [ ] Current interfaces (grpc & rest)
+  - [ ] Current relevant document links
+- [ ] SDD
+  - [ ] Description of grpc & rest handler implementations
+  - [ ] Current sequence diagrams where applicable
+  - [ ] Current relevant document links
+- [ ] CONOPS
+  - [ ] Current information
+  - [ ] Bibliography (if any citations)
+- [ ] README (see svc-cargo for example)
+  - [ ] Badges
+  - [ ] Accurate installation instructions
+  - [ ] Current links to other documents
+  - [ ] Arrow DAO links
+
+## Code (not applicable to examples/ folder)
+- [ ] No unhandled unwrap()s
+- [ ] No ()?; on functions with several exits
```

Comment:
Can you explain reasoning here? Why not use `?` specifically in functions with several exits?

I would say in public functions (library) or request handlers(service) we should "catch" the error, log it and then return it back to user/caller (rethrow). In all inner functions `?` (error propagation) is actually good to use as much as we can? ( within Rust limitations)

Reply:
I don't see any Rust RFC's that state it shouldn't be used in production though, so I would be fine removing this bullet

Reply:
To be clear, I mean this sort of setup:
```rust
fn do_something() -> Result<(), Box<dyn Error>> {
    fxn_1()?;
    // ... 
    fxn_2()?;
    // ...
    fxn_3()?;
}
```

Reply:
Removed this bullet